tonic-build/src/client.rs

```diff
@@ -95,71 +99,91 @@ fn generate_methods(service: &Service, proto: &str) -> TokenStream {
     stream
 }
 
-fn generate_unary(method: &Method, proto: &str, path: String) -> TokenStream {
-    let ident = format_ident!("{}", method.name);
-    let (request, response) = crate::replace_wellknown(proto, &method);
+fn generate_unary<'a, T: Method<'a>>(
+    method: &T,
+    context: &T::Context,
+    path: String,
+) -> TokenStream {
+    let codec_name = format_ident!("{}", T::Context::CODEC_NAME);
+    let ident = format_ident!("{}", method.name());
+    let (request, response) = method.request_response_name(context);
 
     quote! {
         pub async fn #ident(
             &mut self,
             request: impl tonic::IntoRequest<#request>,
         ) -> Result<tonic::Response<#response>, tonic::Status> {
            self.ready().await?;
-           let codec = tonic::codec::ProstCodec::default();
+           let codec = tonic::codec::#codec_name::default();
```

Comment:
I was thinking we should actually make this
```rust
let codec = #codec_path::default();
```
so that you can use any codec from any path include crates that are not tonic.